### PR TITLE
NVIDIA DCGM - New AI Quickstart

### PR DESCRIPTION
**Pull request by RamanaReddy8801**
# Summary

**Ticket**: https://new-relic.atlassian.net/browse/NR-174994
**Description**: Created NVIDIA DCGM quickstart.

A concise description of the changes being introduced. Please review the pre-merge checklist section to validate this pull request is ready for review and merge. **If it is not ready, please mark the pull request as a draft.**

_The owners of this repo are not experts in the subject matter of the quickstarts. We review for the quickstart to be functional and for security risks. If you are seeking feedback on the content of the quickstart, please seek out a subject matter expert. If you are not an internal NR contributor, we can do our best to connect you with a content reviewer._

## Pre merge checklist

- [ ] Did you check you NRQL syntax? - Does it work?
- [ ] Did you include a Data source and Documentation reference?
- [ ] Are all documentation links publicly accessible?
- [ ] Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)?
- [ ] Did you check your descriptive content for spelling and grammar errors?
- [ ] Did you review your content with a subject matter expert? (e.g. a Browser agent quickstart is reviewed with a member of the Browser Agent team)

### Dashboards

- [ ] Does the PR contain a screenshot for each of your dashboards?
- [ ] Do your screenshots show data?
- [ ] Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

### Alerts

- [ ] Did you check that your alerts actually work?

### CODE REVIEW

File: alert-policies/nvidia-dcgm/HighTemperature.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: High GPU Temperature
+
+description: |+
+  This alert is triggered when the Nvidia GPU Temperature is above 90%.
```
Comment (rahulbasu-nr) [inline alert-policies/nvidia-dcgm/HighTemperature.yml:4]:
```suggestion
  This alert is triggered when the NVIDIA GPU Temperature is above 90%.
```

File: dashboards/nvidia-dcgm/nvidia-dcgm.json [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            "row": 1,
+            "width": 2,
+            "height": 2
+          },
+          "linkedEntityGuids": null,
+          "visualization": {
+            "id": "viz.markdown"
+          },
+          "rawConfiguration": {
+            "text": "![NVIDIA DCGM](https://assets.nvidiagrid.net/ngc/logos/DCGM.png)"
```
Comment (rahulbasu-nr) [inline dashboards/nvidia-dcgm/nvidia-dcgm.json:22]: Use the NVIDIA logo from - https://github.com/newrelic/newrelic-quickstarts/assets/104448291/461421da-3f8b-4d71-bac7-2e20d58b4180 Check this in dark theme too.

File: dashboards/nvidia-dcgm/nvidia-dcgm.json [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+                "query": "SELECT average(DCGM_FI_DEV_POWER_USAGE) AS 'usage' FROM Metric WHERE metricName LIKE 'DCGM_FI_DEV_POWER_USAGE' "
+              }
+            ],
+            "platformOptions": {
+              "ignoreTimeRange": false
+            }
+          }
+        },
+        {
+          "title": "Total nvlink bandwidth",
```
Comment (rahulbasu-nr) [inline dashboards/nvidia-dcgm/nvidia-dcgm.json:86]:
```suggestion
          "title": "Total NVLink bandwidth",
```

File: dashboards/nvidia-dcgm/nvidia-dcgm.json [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+            "row": 3,
+            "width": 2,
+            "height": 2
+          },
+          "linkedEntityGuids": null,
+          "visualization": {
+            "id": "viz.markdown"
+          },
+          "rawConfiguration": {
+            "text": "**About**\n\nInstrument your application with New Relic - [Add Data](https://one.newrelic.com).\n\nInstrument NVIDIA DCGM with New Relic using the [documentation](https://docs.newrelic.com/).\n\n[Please rate this dashboard](https://docs.google.com/forms/d/e/1FAIpQLSclR38J8WbbB2J1tHnllKUkzWZkJhf4SrJGyavpMd4t82NjnQ/viewform?usp=pp_url&entry.1615922415=nvidia-dcgm) here and let us know how we can improve it for you."
```
Comment (rahulbasu-nr) [inline dashboards/nvidia-dcgm/nvidia-dcgm.json:128]:
```suggestion
            "text": "**About**\n\nInstrument your application with New Relic - [Add Data](https://one.newrelic.com).\n\nInstrument NVIDIA DCGM with New Relic using the [documentation](https://docs.newrelic.com/docs/infrastructure/host-integrations/host-integrations-list/nvidia-dcgm-integration/).\n\n[Please rate this dashboard](https://docs.google.com/forms/d/e/1FAIpQLSclR38J8WbbB2J1tHnllKUkzWZkJhf4SrJGyavpMd4t82NjnQ/viewform?usp=pp_url&entry.1615922415=nvidia-dcgm) here and let us know how we can improve it for you."
```

File: data-sources/nvidia-dcgm/logo.png
Comment (rahulbasu-nr) [inline data-sources/nvidia-dcgm/logo.png:1]: Use the logo image from https://github-production-user-asset-6210df.s3.amazonaws.com/104448291/279630087-461421da-3f8b-4d71-bac7-2e20d58b4180.png

File: quickstarts/nvidia-dcgm/config.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+description: |
+  ## Why monitor NVIDIA DCGM?
+  monitoring NVIDIA DCGM is essential for maintaining the health and efficiency of your GPU infrastructure in a data center. It helps with performance optimization, fault detection, resource management, energy efficiency, and overall data center health, while also aiding in troubleshooting, security, and compliance.
+  
+  ## Comprehensive monitoring quickstart for NVIDIA DCGM
+  New Relic comprehensive monitoring of your GPU infrastructure in your data center. This setup will allow you to monitor GPU performance and health while leveraging the capabilities of New Relic for data visualization, alerting, and analysis.
+
+  ## What’s included in this quickstart?
+  New Relic NVIDIA DCGM monitoring quickstart provides quality out-of-the-box reporting:
+  - Dashboards (power usage, gpu utilisation, clocks, etc)
```
Comment (rahulbasu-nr) [inline quickstarts/nvidia-dcgm/config.yml:11]:
```suggestion
  - Dashboards (power usage, GPU utilisation, clocks, etc)
```

File: quickstarts/nvidia-dcgm/config.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  - Alerts for ZooKeeper (gpu temperature, xid error)
```
Comment (rahulbasu-nr) [inline quickstarts/nvidia-dcgm/config.yml:12]:
```suggestion
  - Alerts for NVIDIA DCGM (GPU temperature, Xid error)
```

File: quickstarts/nvidia-dcgm/config.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  - New Relic
+  - Ramana Reddy
+title: NVIDIA DCGM
+documentation:
+  - name: NVIDIA DCGM integration documentation
+    description: |
+      Monitor and instrument your NVIDIA DCGM with New Relic to gain deep insights into your performance.
+    url: https://docs.newrelic.com/docs/infrastructure/host-integrations/host-integrations-list/nvidia-dcgm-integration/
+keywords:
+  - NVIDIA DCGM
```
Comment (rahulbasu-nr) [inline quickstarts/nvidia-dcgm/config.yml:29]:
```suggestion
  - NVIDIA DCGM
  - AI Acceleration
  - Machine Learning Acceleration
  - GPU Management
  - AI Management
  - Machine Learning Management
  - Deep Learning Performance
  - AI Performance
  - GPU Optimization
  - AI Optimization

```

File: quickstarts/nvidia-dcgm/logo.png
Comment (rahulbasu-nr) [inline quickstarts/nvidia-dcgm/logo.png:1]: Use the logo image from https://github-production-user-asset-6210df.s3.amazonaws.com/104448291/279630087-461421da-3f8b-4d71-bac7-2e20d58b4180.png

**Review by rahulbasu-nr — CHANGES_REQUESTED** [review]
Please make these changes.

File: data-sources/nvidia-dcgm/config.yml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+id: nvidia-dcgm
+displayName: NVIDIA DCGM
+description: |
+  Monitor and analyze your NVIDIA DCGM infrastructure with New Relic.
+install:
+  primary:
+    link:
+      url: https://docs.newrelic.com/docs/infrastructure/host-integrations/host-integrations-list/nvidia-dcgm-integration/
+icon: logo.png
+keywords:
```
Comment (mickeyryan42) [inline data-sources/nvidia-dcgm/config.yml:10]: For this to show up in the Add data page, you'll need to add a keyword of `NR1_addData`

**Review by mickeyryan42 — CHANGES_REQUESTED** [review]
LGTM, We'll just want to add the keyword i commented separately

**Review by rahulbasu-nr — DISMISSED** [review]
Looks good to me.

**Review by pkosanam — APPROVED** [review]
LGTM

**Review by mickeyryan42 — APPROVED** [review]

**Review by rahulbasu-nr — APPROVED** [review]
Looks good to me.

**Pull request merged**